File: Framework.Repositories/Implementations/Factory/ProjectImageFactory.py
from typing import List, Optional, Any
from datetime import datetime
class ProjectImageFactory:
    """
    Python class equivalent to the C# ProjectImageFactory.
    Likely a repository implementation or Unit of Work.
    """
    # Assuming dependency injection for parameters
    def __init__(self, context: ApplicationDbContext):
        self.context = context

    # TODO: Translate logic from C# method 'GetObj'
    def get_obj(self, id: int) -> ProjectImage: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'SaveObj'
    def save_obj(self, obj: ProjectImage) -> ProjectImage: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'UpdateObj'
    def update_obj(self, obj: ProjectImage) -> ProjectImage: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'GetAll'
    def get_all(self, ) -> List<ProjectImage>: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'GetList'
    def get_list(self, ) -> List<ProjectImage>: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'DeleteObj'
    def delete_obj(self, id: int) -> bool: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'GetImagesByProject'
    def get_images_by_project(self, project_id: int) -> List<ProjectImage>: # Basic return type mapping
        pass

    # TODO: Translate logic from C# method 'GetImagesByProjectBlock'
    def get_images_by_project_block(self, project_block_id: int) -> List<ProjectImage>: # Basic return type mapping
        pass

File: Framework.Repositories/Implementations/Factory/ProjectImageFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Data.Models;
using Framework.Data.Models.Factory;

namespace Data.Repositories
{
	public class ProjectImageFactory
	{
		private readonly ApplicationDbContext _context;

		public ProjectImageFactory(ApplicationDbContext context)
		{
			_context = context;
		}

		#region Crud
		public ProjectImage GetObj(int id)
		{
			return _context.ProjectImages.FirstOrDefault(pi => pi.Id == id);
		}

		public ProjectImage SaveObj(ProjectImage obj)
		{
			_context.ProjectImages.Add(obj);
			_context.SaveChanges();
			return obj;
		}

		public ProjectImage UpdateObj(ProjectImage obj)
		{
			_context.ProjectImages.Update(obj);
			_context.SaveChanges();
			return obj;
		}

		public List<ProjectImage> GetAll()
		{
			return _context.ProjectImages.ToList();
		}

		public List<ProjectImage> GetList()
		{
			return _context.ProjectImages.ToList();
		}

		public bool DeleteObj(int id)
		{
			var obj = GetObj(id);
			if (obj == null)
				return false;
			try
			{
				_context.ProjectImages.Remove(obj);
				_context.SaveChanges();
				return true;
			}
			catch (Exception)
			{
				return false;
			}
		}
		#endregion

		#region Custom Queries
		public List<ProjectImage> GetImagesByProject(int projectId)
		{
			return _context.ProjectImages
						   .Where(pi => pi.ProjectId == projectId)
						   .ToList();
		}

		public List<ProjectImage> GetImagesByProjectBlock(int projectBlockId)
		{
			return _context.ProjectImages
						   .Where(pi => pi.ProjectBlockId == projectBlockId)
						   .ToList();
		}
		#endregion
	}
}
